src/query_bam_records.rs

```diff
@@ -159,7 +160,7 @@ impl BAMQueryRecordReader {
                     second: second_vec,
                 });
                 i += 1;
-            } else { // paired-end
+            } else if self.record_list[i].is_paired() == true && self.record_list.len() % 2 == 0 { // paired-end
```

Comment:
Is it necessary to check `self.record_list[i].is_paired() == true` here? Doesn't `is_paried()` itself return a boolean?